scripts/training_analysis/training_analysis.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+#
+#    This file is part of Leela Zero.
+#    Copyright (C) 2017 Andy Olsen
+#
+#    Leela Zero is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    Leela Zero is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with Leela Zero.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import argparse
+import gzip
+import math
+import fileinput
+
+
+def findEmptyBoard(tfh, hist):
+    first_move_cnt = 0
+    zeros = "0"*math.ceil(19*19/4) + "\n"
+    while (1):
+        #print(tfh.filelineno())
+        empty_board = True
+        board = []
+        for _ in range(16):
+            line = tfh.readline()
+            if not line: break
+            line = line.decode("utf-8")               # Board input planes
+            empty_board = empty_board and line == zeros
+            board.append(line)
+            #print(empty_board, line)
+        if not line: break
+        to_move = int(tfh.readline().decode("utf-8"))           # 0 = black, 1 = white
+        policy_weights = tfh.readline().decode("utf-8")         # 361 moves + 1 pass
+        side_to_move_won = int(tfh.readline().decode("utf-8"))  # 1 = side to move won, -1 = lost
+        # Note: it can be empty_board and white to move if black passed
+        if empty_board and to_move == 0:
+            first_move_cnt += 1
+            policy_weights = policy_weights.split()
+            cnt = policy_weights.count("0")
+            if not cnt in hist: hist[cnt] = 0
```

Comment:
I really like defaultdict(int) for this use case

import collections
hist = collections.defaultdict(int)

hist[cnt] += 1